daemon/discover/main.go

```diff
@@ -108,12 +116,28 @@ func Start(ctx context.Context, drainDuration time.Duration) (stopFunc func(), e
 	<-cfgStarted
 
 	omonStarted := make(chan bool)
+	wg.Add(1)
 	go func(c chan<- bool) {
 		defer wg.Done()
 		d.omon(c)
 	}(omonStarted)
 	<-omonStarted
 
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		d.nodeList.Loop()
+	}()
+
+	// initialize node list
```

Comment:
race move d.nodeList.Add(clusternode.Get()...) to 
	go func() {
 		defer wg.Done()
 		d.nodeList.Add(clusternode.Get()...)           <---
 		d.nodeList.Loop()
 	}()